sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -335,12 +334,80 @@ def formula2(n_features, n_features_to_select, step):
         y = generator.rand(100).round()
         rfecv = RFECV(estimator=SVC(kernel="linear"), step=step, cv=5)
         rfecv.fit(X, y)
-
         assert_equal(rfecv.grid_scores_.shape[0],
                      formula1(n_features, n_features_to_select, step))
         assert_equal(rfecv.grid_scores_.shape[0],
                      formula2(n_features, n_features_to_select, step))
 
+    # Advanced step options
+    generator = check_random_state(0)
+    iris = load_iris()
+    X = np.c_[iris.data, generator.normal(size=(len(iris.data), 20))]
+    y = iris.target
+
+    rfe = RFE(estimator=SVC(kernel="linear"), n_features_to_select=0)
+    assert_raises(ValueError, rfe.fit, X, y)
+    rfe = RFE(estimator=SVC(kernel="linear"), n_features_to_select=10,
+              tune_step_at=10)
+    assert_raises(ValueError, rfe.fit, X, y)
+    rfe = RFE(estimator=SVC(kernel="linear"), step=0)
+    assert_raises(ValueError, rfe.fit, X, y)
+    rfe = RFE(estimator=SVC(kernel="linear"), n_features_to_select=10,
+              tune_step_at=15, tuning_step=0)
+    assert_raises(ValueError, rfe.fit, X, y)
+
+    n_features_list = [300] * 15
+    n_features_to_select_list = [50] * 15
+    step_list = [100, 100, 100, 100, 100, 100,
```

Comment:
I think this would all be clearer if the data was shown zipped in the code (i.e. `(100, 105, 10, False,  [300, 200, 105, 95, 85, 75, 65, 55, 50])` for the first example), perhaps using pytest.mark.parametrize